### DISCUSSION TITLE
Parameter in onColumnSizingChange returns, what seems to be, the setColumnSizing function instead of state

### DISCUSSION THREAD

**Discussion by Tejada-Joyce** [KevinVandy/material-react-table#565] [Q&A] [answered]
When I try to use the value return from `onColumnSizingChange`, I get, what seems to be, the  `setColumnSizing` function instead of the state (`ColumnSizingState`), as the type declaration indicates.

**Type in .d.ts file** 👇 
<img width="596" alt="Screenshot 2023-06-13 at 12 54 36 PM" src="https://github.com/KevinVandy/material-react-table/assets/35240350/f47510d9-cc37-4b1e-b4ac-a586ecf5a1e1">
**Type when trying to use it** 👇 
<img width="589" alt="Screenshot 2023-06-13 at 12 57 36 PM" src="https://github.com/KevinVandy/material-react-table/assets/35240350/0aaadb61-2fb6-4599-8887-5b31ba290312">
<img width="380" alt="Screenshot 2023-06-13 at 1 08 09 PM" src="https://github.com/KevinVandy/material-react-table/assets/35240350/5d7a95ee-f3ee-4fa6-8e4a-8668740603ad">

Is that what I should be getting?

**Comment by KevinVandy** [comment, marked as answer]
This is unfortunately continuing to be one of the most major misconceptions of MRT and TanStack Table. We probably should not have named these apis `on[State]Change`, but rather `set[State]Name`, because that is what they are. They are only meant to work with React useState setStates. 

This is all discussed https://www.material-react-table.com/docs/guides/table-state-management

And if you are wanting to add side effects without useEffect and do stuff beyond a simple react state hoist, I document that near the bottom of the state management docs

https://www.material-react-table.com/docs/guides/table-state-management#add-side-effects-in-set-state-callbacks

```ts
const handleColumnSizingChange = (updater: MRT_Updater<MRT_ColumnSizingState>) => {
  //call the setState as normal, but need to check if using an updater callback with a previous state
  setColumnSizing((prev) =>
    updater instanceof Function ? updater(prev) : updater, //updater is either a raw value or (prev) => callback setter (like react setState)
  );
  //put more code for your side effects here, guaranteed to only run once, even in React Strict Mode
};
```

**Reply by Tejada-Joyce** [reply]
Thank you for your quick reply, Kevin. I was confused because I was using `onColumnOrderChange` and `onDraggingColumnChange`, and they were both "returning" the value (an array of strings and an object, respectively), so I assumed it would be the same for `onColumnSizingChange`, but I just read what you sent, and it makes sense now. Thanks!
